src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift

```diff
@@ -0,0 +1,162 @@
+	let appConfigurationProvider: AppConfigurationProviding
+
+	var supportedCountries: [Country] = []
+
+	var shouldShowSymptomsOnsetScreen = false
+
+	var exposureSubmissionServiceHasRegistrationToken: Bool {
+		exposureSubmissionService.hasRegistrationToken()
+	}
+
+	func symptomsOptionSelected(
+		selectedSymptomsOption: ExposureSubmissionSymptomsViewController.SymptomsOption,
+		isLoading: @escaping (Bool) -> Void,
+		onSuccess: @escaping () -> Void,
+		onError: @escaping (ExposureSubmissionError) -> Void
+	) {
+		switch selectedSymptomsOption {
+		case .yes:
+			shouldShowSymptomsOnsetScreen = true
+			onSuccess()
+		case .no:
+			symptomsOnset = .nonSymptomatic
+			shouldShowSymptomsOnsetScreen = false
+			loadSupportedCountries(isLoading: isLoading, onSuccess: onSuccess, onError: onError)
+		case .preferNotToSay:
+			symptomsOnset = .noInformation
+			shouldShowSymptomsOnsetScreen = false
+			loadSupportedCountries(isLoading: isLoading, onSuccess: onSuccess, onError: onError)
+		}
+	}
+
+	func symptomsOnsetOptionSelected(
+		selectedSymptomsOnsetOption: ExposureSubmissionSymptomsOnsetViewController.SymptomsOnsetOption,
+		isLoading: @escaping (Bool) -> Void,
+		onSuccess: @escaping () -> Void,
+		onError: @escaping (ExposureSubmissionError) -> Void
+	) {
+		switch selectedSymptomsOnsetOption {
+		case .exactDate(let date):
+			guard let daysSinceOnset = Calendar.gregorian().dateComponents([.day], from: date, to: Date()).day else { fatalError("Getting days since onset from date failed") }
+			symptomsOnset = .daysSinceOnset(daysSinceOnset)
+		case .lastSevenDays:
+			symptomsOnset = .lastSevenDays
+		case .oneToTwoWeeksAgo:
+			symptomsOnset = .oneToTwoWeeksAgo
+		case .moreThanTwoWeeksAgo:
+			symptomsOnset = .moreThanTwoWeeksAgo
+		case .preferNotToSay:
+			symptomsOnset = .symptomaticWithUnknownOnset
+		}
+
+		loadSupportedCountries(isLoading: isLoading, onSuccess: onSuccess, onError: onError)
+	}
+
+
+	func warnOthersConsentGiven(
+		isLoading: @escaping (Bool) -> Void,
+		onSuccess: @escaping () -> Void,
+		onError: @escaping (ExposureSubmissionError) -> Void
+	) {
+		startSubmitProcess(
+			isLoading: isLoading,
+			onSuccess: onSuccess,
+			onError: onError
+		)
+	}
+
+	// MARK: - Private
+
+	private var symptomsOnset: SymptomsOnset = .noInformation
+	private var consentToFederationGiven: Bool = false
+
+	private func loadSupportedCountries(
+		isLoading: @escaping (Bool) -> Void,
+		onSuccess: @escaping () -> Void,
+		onError: @escaping (ExposureSubmissionError) -> Void
+	) {
+		isLoading(true)
+		appConfigurationProvider.appConfiguration { applicationConfiguration in
+			isLoading(false)
+
+			guard let supportedCountries = applicationConfiguration?.supportedCountries.compactMap({ Country(countryCode: $0) }) else {
+				onError(.noAppConfiguration)
+				return
+			}
+
+			if supportedCountries.isEmpty {
+				self.supportedCountries = ["DE"].compactMap { Country(countryCode: $0) }
```

Comment:
would DE also be available via Country.defaultCountry() ?

Reply:
Thanks, that's right, totally makes sense. Replaced it 👍